guides/k8-akka-cluster/src/main/paradox/index.md

```diff
@@ -0,0 +1,600 @@
+* An existing clustered Akka based application to deploy, built using SBT or Maven.
+
+## Overview
+
+First, we will need to containerize the application and publish it to the Docker registry used by the Kubernetes cluster. This guide will show how to configure both SBT and Maven to perform this task.
+
+Once our image is published, we will utilize Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) to deploy the application. Using StatefulSet, given a service named `myapp` and `3` replicas, Kubernetes will start `3` [Pods](https://kubernetes.io/docs/concepts/workloads/pods/pod/) with the names `myapp-0`, `myapp-1`, and `myapp-2`. These Pod names will be registered in the Kubernetes DNS, such that they can be resolved by the pods within the same StatefulSet. This would mean `myapp-0` as a host name can be resolved within the `myapp-2` pod, for example.
+
+Based on the above example, the container `myapp-0` can be used as the seed node to form the Akka cluster within Kubernetes.
+
+Each Pod will also need to expose the Akka remoting port so it is accessible from a different Pod instance. We will utilize Kubernetes [Service](https://kubernetes.io/docs/concepts/services-networking/service/) to achieve this.
+
+Our overall steps will be:
+
+1. [Naming your application](Naming-your-application)
+2. [Handling environment variables](Handling-environment-variables)
+3. [Containerizing your application](Containerizing-your-application)
+3.1 [Containerizing your application - SBT](Containerizing-your-application---SBT)
+3.1.1 [SBT multi-module project](SBT-multi-module-project)
+3.1.2 [SBT single-module project](SBT-single-module-project)
+3.1.3 [Optional: Using a smaller Docker base image](Optional:-Using-a-smaller-Docker-base-image)
+3.2 [Containerizing your application - Maven](Containerizing-your-application---Maven)
+3.2.1 [Maven multi-module project](Maven-multi-module-project)
+3.2.2 [Maven single-module project](Maven-single-module-project)
+4. [Creating Kubernetes Service for Akka remoting](Creating-Kubernetes-Service-for-Akka-remoting)
+5. [Creating Kubernetes StatefulSet resource](Creating-Kubernetes-StatefulSet-resource)
+
+
+
+## 1. Naming your application
+
+In this guide we will refer to the application name as `myapp` - please feel free to substitute the application name with the actual name of the application you are going to deploy.
+
+We'd like to bring your attention to the application name as they would be used to define the service name in the Kubernetes StatefulSet and Kubernetes Service.
+
+The application name will also be referenced in the SBT or Maven build file, and as such the application name will form the Docker image name when containerized.
+
+
+## 2. Handling environment variables
+
+To establish Akka cluster within the Kubernetes container, we will be adding the following system properties when containerizing your application:
```

Comment:
To establish an Akka cluster within the Kubernetes container, a later section will show you how to add the following system properties necessary for containerizing your application: